futures-util/src/stream/futures_unordered/mod.rs

```diff
@@ -49,6 +49,9 @@ pub struct FuturesUnordered<Fut> {
     ready_to_run_queue: Arc<ReadyToRunQueue<Fut>>,
     len: usize,
     head_all: *const Task<Fut>,
+    /// Track whether we have yielded `None` and can consider ourselves
+    /// terminated
+    is_terminated: bool,
```

Comment:
I'd really like to figure out how to pack this into the same size somehow. Maybe we could set `len` to `usize::MAX` or something as a discriminant? Seems funky, but might work okay.

Reply:
Yep, I think that should be relatively straightforward.